### PR TITLE
Fix framework-arduinoespressif32 issue

### LINKED ISSUE #189: Build failure with latest arduino-esp32 sources
Tracking from #185: A clean build fails with the following errors:

```
$ pio run -e d32_pro_tft
[...]
Tool Manager: Installing git+https://github.com/espressif/arduino-esp32.git
git version 2.31.0
Cloning into '/Users/me/.platformio/.cache/tmp/pkg-installing-c8n23ck4'...
remote: Enumerating objects: 3626, done.
remote: Counting objects: 100% (3626/3626), done.
remote: Compressing objects: 100% (2645/2645), done.
remote: Total 3626 (delta 892), reused 2080 (delta 537), pack-reused 0
Receiving objects: 100% (3626/3626), 75.29 MiB | 6.74 MiB/s, done.
Resolving deltas: 100% (892/892), done.
Updating files: 100% (4354/4354), done.
Tool Manager: framework-arduinoespressif32 @ 0.0.0+sha.371f382 has been installed!
[...]
.pio/libdeps/d32_pro_tft/LCBUrl/src/LCBUrl.cpp: In member function 'IPAddress LCBUrl::getIP(const char*)':
.pio/libdeps/d32_pro_tft/LCBUrl/src/LCBUrl.cpp:634:53: error: cannot convert 'ip4_addr*' to 'esp_ip4_addr_t* {aka esp_ip4_addr*}' for argument '3' to 'esp_err_t mdns_query_a(const char*, uint32_t, esp_ip4_addr_t*)'
         esp_err_t err = mdns_query_a(hn, 2000, &addr);
                                                     ^
.pio/libdeps/d32_pro_tft/WiFiManager/WiFiManager.cpp: In member function 'void WiFiManager::WiFiEvent(arduino_event_id_t, system_event_info_t)':
.pio/libdeps/d32_pro_tft/WiFiManager/WiFiManager.cpp:3477:17: warning: comparison between 'enum arduino_event_id_t' and 'enum system_event_id_t' [-Wenum-compare]
     if(event == SYSTEM_EVENT_STA_DISCONNECTED){
                 ^
.pio/libdeps/d32_pro_tft/WiFiManager/WiFiManager.cpp:3494:20: warning: comparison between 'enum arduino_event_id_t' and 'enum system_event_id_t' [-Wenum-compare]
   else if(event == SYSTEM_EVENT_SCAN_DONE){
                    ^
.pio/libdeps/d32_pro_tft/WiFiManager/WiFiManager.cpp: In member function 'void WiFiManager::WiFi_autoReconnect()':
.pio/libdeps/d32_pro_tft/WiFiManager/WiFiManager.cpp:3511:79: error: no matching function for call to 'WiFiClass::onEvent(std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type)'
       wm_event_id = WiFi.onEvent(std::bind(&WiFiManager::WiFiEvent,this,_1,_2));
                                                                               ^
In file included from /Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFiSTA.h:28:0,
                 from /Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFi.h:32,
                 from .pio/libdeps/d32_pro_tft/WiFiManager/WiFiManager.h:63,
                 from .pio/libdeps/d32_pro_tft/WiFiManager/WiFiManager.cpp:13:
/Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFiGeneric.h:145:21: note: candidate: wifi_event_id_t WiFiGenericClass::onEvent(WiFiEventCb, arduino_event_id_t)
     wifi_event_id_t onEvent(WiFiEventCb cbEvent, arduino_event_id_t event = ARDUINO_EVENT_MAX);
                     ^
/Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFiGeneric.h:145:21: note:   no known conversion for argument 1 from 'std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type {aka std::_Bind<std::_Mem_fn<void (WiFiManager::*)(arduino_event_id_t, system_event_info_t)>(WiFiManager*, std::_Placeholder<1>, std::_Placeholder<2>)>}' to 'WiFiEventCb {aka void (*)(arduino_event_id_t)}'
/Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFiGeneric.h:146:21: note: candidate: wifi_event_id_t WiFiGenericClass::onEvent(WiFiEventFuncCb, arduino_event_id_t)
     wifi_event_id_t onEvent(WiFiEventFuncCb cbEvent, arduino_event_id_t event = ARDUINO_EVENT_MAX);
                     ^
/Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFiGeneric.h:146:21: note:   no known conversion for argument 1 from 'std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type {aka std::_Bind<std::_Mem_fn<void (WiFiManager::*)(arduino_event_id_t, system_event_info_t)>(WiFiManager*, std::_Placeholder<1>, std::_Placeholder<2>)>}' to 'WiFiEventFuncCb {aka std::function<void(arduino_event_id_t, arduino_event_info_t)>}'
/Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFiGeneric.h:147:21: note: candidate: wifi_event_id_t WiFiGenericClass::onEvent(WiFiEventSysCb, arduino_event_id_t)
     wifi_event_id_t onEvent(WiFiEventSysCb cbEvent, arduino_event_id_t event = ARDUINO_EVENT_MAX);
                     ^
/Users/me/.platformio/packages/framework-arduinoespressif32/libraries/WiFi/src/WiFiGeneric.h:147:21: note:   no known conversion for argument 1 from 'std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type {aka std::_Bind<std::_Mem_fn<void (WiFiManager::*)(arduino_event_id_t, system_event_info_t)>(WiFiManager*, std::_Placeholder<1>, std::_Placeholder<2>)>}' to 'WiFiEventSysCb {aka void (*)(arduino_event_t*)}'
At global scope:
cc1plus: warning: unrecognized command line option '-Wno-frame-address'
Compiling .pio/build/d32_pro_tft/lib236/Wire/Wire.cpp.o
*** [.pio/build/d32_pro_tft/lib9fe/LCBUrl/LCBUrl.cpp.o] Error 1
At global scope:
cc1plus: warning: unrecognized command line option '-Wno-frame-address'
*** [.pio/build/d32_pro_tft/libbe0/WiFiManager/WiFiManager.cpp.o] Error 1
```

### PR DESCRIPTION
**Pull request by ricnewton**
Latest master branch has broken some libraries, pin to just prior to breaking changes for now.

Resolves #189

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 [common]
 platform = espressif32
-platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git
+platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#46d5afb17fb91965632dc5fef237117e1fe947fc
```
Comment (TaplistIOTeam) [inline platformio.ini:19]: Suggest updating this to:

```
platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#1.0.6
```

(It's the tag for the same hash/release - just a tad more at-a-glance understandable).

BTW, I've verified that this fixes #189. Feel free to add a `Resolves #189` to your PR desc to auto-close upon merge.

**Comment by TaplistIOTeam** [comment]
`/me` anxiously watches the [github actions run](https://github.com/thorrak/tiltbridge/runs/2327792463?check_suite_focus=true) 😅

File: platformio.ini (hunk 1 above)
Comment (ricnewton) [reply platformio.ini]: OK, done.

**Comment by ricnewton** [comment]
All green :)

**Comment by TaplistIOTeam** [comment]
Nice!!

**Comment by TaplistIOTeam** [comment]
Tested locally, confirmed works as well.

@lbussy I think this is the last thing necessary to turn `devel` green again.

**Comment by lbussy** [comment]
I can merge it, but I don't understand it. :)

@thorrak?

**Comment by thorrak** [comment]
> I can merge it, but I don't understand it. :)
> 
> @thorrak?

I mean... The answer is somewhere in [here](https://github.com/espressif/arduino-esp32/compare/1.0.6...master). 

> Showing 5,236 changed files with 843,750 additions and 340,175 deletions.

Yeeaahhhh. They bumped the underlying framework from 1.0.6 to 2.0. I'm going to vote we merge this commit, stick with the 1.0.x branch for now, and worry about upgrading to 2.0.x _much_ later.

**Comment by ricnewton** [comment]
@lbussy So commit https://github.com/espressif/arduino-esp32/commit/5502879a5b25e5fff84a7058f448be481c0a1f73 in the arduino-esp32 repo changed a load of apis (including e.g. mdns_query_a) that libraries LCBUrl and WiFiManager need, so a completely clean build will currently fail if arduino-esp32 is post that commit.

It seems like some sort of caching in platform io hides this sometime as @TaplistIOTeam had to remove his .platformio directory in the home directory to be able to reproduce the failure.

**Comment by ricnewton** [comment]
@lbussy FYI I got LIBUrl working by changing ``struct ip4_addr addr;`` to ``esp_ip4_addr_t addr;`` in ``LCBUrl::getIP``, but then WifiManager still fails

**Comment by TaplistIOTeam** [comment]
> It seems like some sort of caching in platform io hides this sometime as @TaplistIOTeam had to remove his .platformio directory in the home directory to be able to reproduce the failure.

That's right; as is, the dependency is expressed as "repo @ master"; which of course will have a different value depending on when evaluated. It seems platformio sees a checked-out repo@master and decides that's good enough.

Arguably, it's a good general practice even in the absence of build breakage to pin third-party dependencies to a specific version, so that builds are reproducible and "hermetic": The same checkout from our tree is guaranteed to build against the same total set of sources, and yields the same object code. Further, you have the auditability benefit that dependency changes are always logged explicitly, by a commit/pr. 

(You'll see this style of dependency pinning in other ecosystems, eg go's modules & node's `package-lockfile.json`, etc. It's of course up to you whether to adopt it here.)

**Pull request merged**